.github/workflows/android-release.yml

```diff
@@ -0,0 +1,83 @@
+#This is a workflow that creates, downloads, uploads and releases APK-file.
+name: APK Workflow
+
+#Event triggers
+#execute the jobs everytime when a pull_request to the main-branch is created
+on:
+  push:
+    branches: main
+
+jobs:
+  runtests:
+
+    name: Run Setup Tests
+    runs-on: ubuntu-latest
+    steps:
+
+      #Checkout a Git repository and set up Java JDK
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      #command: run unit tests
+      - name: Run Build Test
+        #Working directory is the folder with the project
+        working-directory: ./BibBuddy
+        run: bash ./gradlew test --stacktrace
+
+      #command:  Run Java Checkstyle
+      - name: Run Java Checkstyle
+        # You may pin to the exact commit or the version.
+        # uses: nikitasavinov/checkstyle-action@d236d4d9a3563684368581f1127f7e4773013e9d
+        uses: nikitasavinov/checkstyle-action@0.3.1
+        with:
+          checkstyle_config: google_checks.xml
+          tool_name: Java Checkstyle
+
+  apk:
+    name: APK
+
+    #to run this job, the job build_test should be successful  
+    #otherwise: this job will be skipped
+    needs: runtests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Build APK
+        working-directory: ./BibBuddy
+        run: bash ./gradlew assembleDebug --stacktrace
+
+      #Upload the APK as a build artifact that can be used by subsequent workflow steps
+      - name: Upload APK
+        uses: actions/upload-artifact@v2.2.2
+        with:
+          name: apk_file
+          path: ./BibBuddy/app/build/outputs/apk/debug/app-debug.apk 
+
+      #Download a build artifact that was previously uploaded in the workflow by the upload-artifact action
+      - name: Download APK
+        uses: actions/download-artifact@v2.0.8
+        with:
+          name: apk_file
+
+      #Create a release for a tag in your repository           
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1.1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+        with:
+          #The name of the tag for this release
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          body: This is a release of the created APK
```

Comment:
The release is created but the created release is not uploaded yet?

Reply:
The release is uploaded as a part of the artifact. Everyone can download the created apk:
![image](https://user-images.githubusercontent.com/37459401/109420542-5b321000-79d3-11eb-8416-c4b153880524.png)

See this part of the .yml

```
      #Upload the APK as a build artifact that can be used by subsequent workflow steps
      - name: Upload APK
        uses: actions/upload-artifact@v2.2.2
        with:
          name: apk_file
          path: ./BibBuddy/app/build/outputs/apk/debug/app-debug.apk 

```

Reply:
Ok, should the APK also be released in Google Play Store with our bibBuddy account?

Reply:
yes, I think so.